AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/LocalStorage/LocalStorageProtocol.swift

```diff
@@ -0,0 +1,18 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import SQLite
+
+protocol LocalStorageProtocol {
+    /// Executes a SQL statement
+    /// - Parameters:
+    ///   - statement: SQL statement
+    ///   - bindings: Collection of SQL Bindings
+    /// - Returns: A Single SQL Statement
+    func executeSqlQuery(_ statement: String, _ bindings: [Binding?]) throws -> Statement
```

Comment:
since `LocalStorageProtocol` seems to be agnostic in respect to the underlying storage, can we perhaps change this method name such that doesn't have any reference to `SQL` and the `SQLite` library?

Reply:
note: I'd also add a generic "initializeStorage"-like method that types conforming to this protocol should implement to inject their own logic (i.e. creating the db and tables for SQLite or something for a different adapter)

Reply:
I've re-organize the structure a little bit.  I've made the LocalStorageProtocol specific to SQL and renamed it to SQLStorageProtocol where the adapter is limited to creating table and executing SQL queries and doesn't know anything about Events or Pinpoint Events.  The agnostic abstraction is moved one layer up to the AnalyticsEventStorage and AnalyticsEventSQLStorage implementation.  I'm open to deleting and removing this protocol completely if it doesn't make sense anymore.